enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,685 @@
+
+### High Level View of Kueue
+
+Kueue's [website](https://kueue.sigs.k8s.io/docs/overview/) provides a list of their functionality.
+
+Kueue provides the following functionality:
+
+- Job Management
+- Advanced Resource Management
+- Integrations with common workloads
+- Advanced Autoscaling support
+- All-or-nothing with Ready Pods
+- Partial Admission and dynamic reclaim
+- Mixing Training and Inferencing
+- Multi-cluster job dispatching
+- Topology-Aware Scheduling
+
+Kueue uses [Resource Flavors](https://kueue.sigs.k8s.io/docs/concepts/resource_flavor/) to describe what resources are available on a cluster to better support heterogenous clusters. 
+Resource flavors are cluster scoped and set up by the cluster adminstrator.
+
+Kueue uses [Cluster Queues](https://kueue.sigs.k8s.io/docs/concepts/cluster_queue/) to governs a pool of resources, defining usage limits and fair sharing rules.
+Cluster Queues are cluster scoped. Cluster queues are set up by a cluster administrator.
+
+Kueue uses [Local Queues](https://kueue.sigs.k8s.io/docs/concepts/local_queue/) to group closely related workloads belonging to a single tenant.
+LocalQueues are namespaced scoped and they link with the cluster queues that a user can use for their workloads.
+
+Internally Kueue uses a concept of a [Workload](https://kueue.sigs.k8s.io/docs/concepts/workload/) to translate k8s objects into a unit of admission.
+Workload API is what Kueue uses to enforce its quota logic.
+All supported frameworks of Kueue translate the objects into a Workload.
+
+Kueue provides a choice of integration via either dedicated frameworks or via the use of pod scheduling gates.
+Many support frameworks use the Suspend field and Kueue will unsuspend workloads once their is capacity in their cluster.
+Kueue will typically look at all namespaces of a cluster and check if that workload should be gated by Kueue.
+
+The use of pod scheduling gates is another approach that Kueue uses to enforce quota management. 
+In this case, Kueue will add a scheduling gate if the workload is quota limited. This will gate the pod and kueue will patch the pod to release the gate.
+Once that happens the pod will be scheduled as normal.
+The use of the pod scheduling gate enables integrations such as Deployments, Statefulsets and LeaderWorkerSet.
+
+Kueue liberally uses webhooks to patch workloads depending on if the namespace is labeled to support Kueue or the workload has a queue name label.
+
+The major challange of Kueue for Openshift is configuration and support of their various frameworks and these different approaches.
+
+### User Stories
+
+#### Machine Learning Serving
+
+As a LLM serving provider, I want to use Kueue to provide quota management for serving workloads. 
+Cluster admins can limit access to GPUs so that a single user won’t use all the GPUs on the cluster.
+
+Kueue provides workload support for StatefulSet, Deployments and LeaderWorkerSet. 
```

Comment:
Is this still going to be considered as a user story for the enhancement? Based on previous conversations it was brought up that OCP team will not support SS, Deployments, Pods and LWS as a part of Kueue migration story.

Reply:
Maybe it isn't as clear. We see two focus points: production usage and experimental usage.

For RHOAI we know that there is going to be loads of dev preview/POC work so we want to offer a experimental option for the operator where support may be limited from OCP but it would provide you all the ability to POC features. We are still figuring out the best option for offering an expermental channel for the operator but we know that we want to provide more configuration to advanced users but we want to limit Production to a subset of supported options for now.

As of now I don't know of anyone in OCP that has a design figured out for pod integration so we don't think it will be part of our default settings for integrations. But that doesn't mean we won't support it.

Reply:
I agree. I think we need to sync up on pod integration in OCP.

I tend to lead towards an opt-in approach based on a namespace label.

Reply:
I'm going to resolve this one but we will continue to discuss this offline.